src/model/keychain.ts

```diff
@@ -262,3 +273,69 @@ export async function getPrivateAccessControlOptions(): Promise<Options> {
 
   return res;
 }
+
+// If user does a backup of an iPhone, wipes it and restores the backup on the
+// same phone, keys created using secure enclave will be missing, but due to the
+// flag AccessibleWhenUnlockedThisDeviceOnly, public keys will be restored.
+// This function is responsible for:
+// 1. Figuring out if it is possible that wallet is in a broken state (empty
+//    local storage and available public keychain).
+// 2. If there is a reasonable suspicion that keychain might be broken, we check
+//    if there are seed phrases (this invokes FaceID check) and if they are
+//    valid.
+// Result is true if keychain is needs a repair, false otherwise.
+export async function checkKeychainStatus() {
```

Comment:
This is a nit and probably it's not necessary at all, so do whatever you think is best.

Perhaps you could return string status from `checkKeychainStatus` instead of boolean with a comment. 
```
type KeychainStatus = 'operational' | 'addressMissing' | 'storageEmpty' | 'failedToLoad'
```

I'm. not 100% sure about it, but it may make the code a little bit more self documenting in terms of statuses.

Reply:
All of this doesn't matter. We are only interested in answer to the question if keychain needs to be repaired.